chain-config/src/validator_rules.rs

```diff
@@ -40,22 +52,22 @@ pub trait ValidatorRulesModule: setup_phase::SetupPhaseModule + events::EventsMo
 
         self.last_bls_key_id().set(current_bls_key_id.clone());
         self.bls_keys_map()
-            .insert(current_bls_key_id.clone(), new_validator.bls_key.clone());
-        self.bls_key_to_id_mapper(&new_validator.bls_key)
+            .insert(current_bls_key_id.clone(), new_bls_key.clone());
+        self.bls_key_to_id_mapper(&new_bls_key)
             .set(current_bls_key_id.clone());
 
         self.register_event(
             &current_bls_key_id,
-            &new_validator.address,
-            &new_validator.bls_key,
-            &new_validator.egld_stake,
-            &new_validator.token_stake,
+            &self.blockchain().get_caller(),
+            &new_bls_key,
+            &egld_stake,
+            &additional_stake,
         );
     }
 
+    // TODO: add storage for registered stake in order to return it upon unregistering
```

Comment:
We said we can save the registered stake into ValidatorInfo and set it in the MapMapper